### PR TITLE
OPERA: _WD_UpdateDriver + __WD_UpdateExtractor

### PR DESCRIPTION
**Pull request by mlipok**
### Proposed changes

Support for unpacking opera driver which is placed in subfolder in zip archive

### Checklist

Put an `x` in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We are here to help!<br>
This is simply a reminder of what we are going to look for before merging your code.

- [x] I have read and noticed the [CODE OF CONDUCT](https://github.com/Danp2/au3WebDriver/blob/master/docs/CODE_OF_CONDUCT.md) document
- [x] I have read and noticed the [CONTRIBUTING](https://github.com/Danp2/au3WebDriver/blob/master/docs/CONTRIBUTING.md) document
- [ ] I have added necessary documentation or screenshots (if appropriate)

### Types of changes

Please check `x` the type of change your PR introduces:

- [ ] Bugfix (change which fixes an issue)
- [x] Feature (change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (functional, structural)
- [ ] Documentation content changes
- [ ] Other (please describe)

### What is the current behavior?

it is not able to unpack opera driver

### What is the new behavior?

support for unpacking exe file from subfolder

### Additional context

https://github.com/Danp2/au3WebDriver/issues/187

### System under test

NOT RELATED

### CODE REVIEW

File: wd_helper.au3 [hunk 1]
```diff
@@ -1345,53 +1350,15 @@ Func _WD_UpdateDriver($sBrowser, $sInstallDir = Default, $bFlag64 = Default, $bF
...
-						#EndRegion - Extract new instance of webdriver
+						; Extract
+						__WD_UpdateExtractor($sTempFile, $sInstallDir, $sDriverEXE)
+						If Not @error Then $bResult = True
+						$iErr = @error
+						$iExt = @extended
 					EndIf
 					FileDelete($sTempFile)
+					ShellExecute($sInstallDir)
+					Exit
```
Comment (mlipok) [inline wd_helper.au3:1361]: `ShellExecute` and `Exit` are only for easier testing together with `wd_demoa.u3`

Should be deleted before merge to master

File: wd_helper.au3 [hunk 2]
```diff
@@ -1410,6 +1377,93 @@ Func _WD_UpdateDriver($sBrowser, $sInstallDir = Default, $bFlag64 = Default, $bF
...
+			For $FileItem In $FilesInZip     ; Check the files in the archive separately
+				; https://docs.microsoft.com/en-us/windows/win32/shell/folderitem
+
+				; ConsoleWrite only for testing - should be deleted before merge to master
+				ConsoleWrite("! " & @ScriptLineNumber & ' ' & $FileItem.Name & @CRLF)
+				ConsoleWrite("! " & @ScriptLineNumber & ' ' & $FileItem.Type & @CRLF)
+				ConsoleWrite("! " & @ScriptLineNumber & ' ' & $FileItem.IsFolder & @CRLF)
+				ConsoleWrite("! " & @ScriptLineNumber & ' ' & IsBool($FileItem.IsFolder) & @CRLF)
+				ConsoleWrite("! " & @ScriptLineNumber & ' ' & IsString($FileItem.IsFolder) & @CRLF)
+				ConsoleWrite("! " & @ScriptLineNumber & ' ' & VarGetType($FileItem.IsFolder) & @CRLF)
```
Comment (mlipok) [inline wd_helper.au3:1434]: ConsoleWrite only for testing - should be deleted before merge to master

File: wd_helper.au3 [hunk 3]
```diff
@@ -1410,6 +1377,93 @@ Func _WD_UpdateDriver($sBrowser, $sInstallDir = Default, $bFlag64 = Default, $bF
...
+;                  $sDriverEXE          - a string value.
+;                  $sSubDir             - [optional] a string value. Default is "".
+; Return values .: None
+; Author ........: Danp2
+; Modified ......: mLipok
+; Remarks .......:
+; Related .......:
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
```
Comment (mlipok) [inline wd_helper.au3:1395]: @Danp2  please complete the function header

File: wd_helper.au3 [hunk 4]
```diff
@@ -1302,10 +1302,15 @@ Func _WD_UpdateDriver($sBrowser, $sInstallDir = Default, $bFlag64 = Default, $bF
...
 
 		If $iErr = $_WD_ERROR_Success Then
-			; Match exe file name in list of supported browsers
+			If StringInStr($sBrowser, '\') Then
+				; Extract filename from full path
+				$sBrowser = StringRegExpReplace($sBrowser, "^.*\\|\..*$", "")
+			EndIf
+
+			; Match in list of supported browsers
+			Local $iIndex = _ArraySearch($_WD_SupportedBrowsers, $sBrowser, Default, Default, Default, Default, Default, $_WD_BROWSER_Name)
```
Comment (Danp2) [inline wd_helper.au3:1313]: I'm not understanding why this needed to be changed. Did the earlier usage of `$iIndex = @extended` not work? Please explain.

Comment (mlipok) [reply wd_helper.au3:1313]: oops ... cross version error.
Will fix this soon.

Comment (mlipok) [reply wd_helper.au3]: Take a look: [StringInStr($sBrowser, '\') and $iIndex --- fixed](https://github.com/Danp2/au3WebDriver/pull/236/commits/a23eb7f278559a77ef389e3a2705628dd5b243c3)

File: wd_helper.au3 [hunk 5]
```diff
@@ -1345,53 +1345,17 @@ Func _WD_UpdateDriver($sBrowser, $sInstallDir = Default, $bFlag64 = Default, $bF
...
+						__WD_UpdateExtractor($sTempFile, $sInstallDir, $sDriverEXE)
+						If Not @error Then $bResult = True
+						$iErr = @error
+						$iExt = @extended
 					EndIf
 					FileDelete($sTempFile)
+					; ShellExecute and Exit are only for easier testing together with wd_demo.au3
+					; Should be deleted before merge to master
+					; ShellExecute($sInstallDir)
+					; Exit
```
Comment (mlipok) [inline wd_helper.au3:1358]: Should be deleted before merge to master

**Review by Danp2 — APPROVED** [review]

**Pull request merged**